### PR TITLE
Custom tokens

### PR DESCRIPTION
**Pull request by ignaciosantise**
## Summary
* Added custom tokens functionality
* Fully-refactored Text Input component
* Code improvements
* Migrated common components to TS

### Android
https://user-images.githubusercontent.com/25931366/185636754-81aaefad-fa5c-4498-8691-9e4c92cc01f1.mp4

### iOS
https://user-images.githubusercontent.com/25931366/185636864-f03a1a86-5cb2-4782-981f-8d0ce922a3c3.MP4

## Github Issue
https://github.com/Psychedelic/plug-mobile/issues/393

### CODE REVIEW

File: src/screens/tabs/Profile/screens/Settings/components/SettingItem/index.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ const SettingItem = ({ icon, name, description, onPress, iconName }) => (
         <Text style={[FontStyles.Normal, styles.name]}>{name}</Text>
         <Text style={FontStyles.NormalGray}>{description}</Text>
       </Column>
-      <Icon name="chevronRight" style={styles.chevron} />
+      <Icon name="chevronRight" style={styles.chevron} height={18} />
```
Comment (mattgle) [inline src/screens/tabs/Profile/screens/Settings/components/SettingItem/index.js:23]: why not height inside the  style?

File: src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+interface Props {
+  modalRef: React.RefObject<Modalize>;
+  onSelectedToken: (token: DABToken) => void;
+}
+
+function CustomToken({ modalRef, onSelectedToken }: Props) {
+  const [canisterId, setCanisterId] = useState('');
+  const dispatch = useDispatch();
+  const [standard, setStandard] = useState<FungibleStandard>('DIP20');
```
Comment (mattgle) [inline src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx:36]: DIP20 should be a value from a enum or something like that

File: src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  const [loading, setLoading] = useState(false);
+  const [canisterIdError, setCanisterIdError] = useState(false);
+  const [tokenError, setTokenError] = useState<Error>();
+  const optionsRef = useRef<Modalize>(null);
+  const showCanisterError =
+    tokenError?.type === 'INVALID_CANISTER_ID' ||
+    tokenError?.type === 'INVALID_NFT';
+  const showStandardError =
+    tokenError?.type === 'INVALID_STANTARD' || tokenError?.type === 'DEFAULT';
```
Comment (mattgle) [inline src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx:45]: enum of the types?

File: src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+          buttonStyle={[
+            styles.standardButton,
+            showStandardError && styles.standardButtonError,
+          ]}
+          textStyle={
+            standard ? styles.standardText : styles.standardTextPlaceholder
+          }
+          iconStyle={styles.standardIcon}
+          iconName="chevronRight"
+          iconProps={{ height: 18 }}
```
Comment (mattgle) [inline src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx:170]: extract into styles

File: src/screens/tabs/Profile/screens/Settings/components/SettingItem/index.js (hunk 1 above)
Comment (ignaciosantise) [reply src/screens/tabs/Profile/screens/Settings/components/SettingItem/index.js:23]: it doesnt work in SVGs, they need to be sent as props

File: src/screens/tabs/Tokens/components/AddToken/steps/TokenList/index.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    Keyboard.dismiss();
+    modalRef?.current?.open();
+  };
+
+  const handleOpenDabForm = useCallback(() => {
+    Linking.canOpenURL(dabFormUrl).then(() => Linking.openURL(dabFormUrl));
+  }, []);
+
+  useEffect(() => {
+    if (search === '') {
```
Comment (mattgle) [inline src/screens/tabs/Tokens/components/AddToken/steps/TokenList/index.tsx:62]: Does this works?:

```suggestion
    if (search) {
```

**Review by mattgle — COMMENTED** [review]
Looking good, just a few comments

File: src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx (hunk 3 above)
Comment (ignaciosantise) [reply src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx:45]: they are defined as types
Screen Shot 2022-08-22 at 12 38 10
Screen Shot 2022-08-22 at 12 38 15

File: src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx (hunk 2 above)
Comment (ignaciosantise) [reply src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx:36]: same, they are already defined as types in `FungibleStandard`

Comment (ignaciosantise) [reply src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx:36]: Screen Shot 2022-08-22 at 12 39 30

File: src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx (hunk 4 above)
Comment (ignaciosantise) [reply src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx:170]: the thing here is that they are not styles, they need to be sent as an object. 
In the future i would like to remove that icon logic and directly send the icon as prop like this:

`left={<Icon ... />}

File: src/screens/tabs/Tokens/components/AddToken/steps/TokenList/index.tsx (hunk 5 above)
Comment (ignaciosantise) [reply src/screens/tabs/Tokens/components/AddToken/steps/TokenList/index.tsx:62]: cool

**Pull request merged**